conda/exceptions.py

```diff
@@ -1323,6 +1323,32 @@ def __init__(
         super().__init__(msg, *args, **kwargs)
 
 
+class EnvironmentExporterNotDetected(CondaError):
+    def __init__(
+        self,
+        filename: str,
+        available_formats: Iterable[str],
+        supported_filenames: Iterable[str],
+        *args,
+        **kwargs,
+    ):
+        self.filename = filename
+
+        msg = dals(
+            f"""
+            No environment exporter plugin found for filename '{filename}'.
+
+            Supported filenames: {dashlist(supported_filenames, 16)}
+
+            Available formats: {dashlist(available_formats, 16)}
+
+            Use conda export --format=FORMAT to specify the export format explicitly,
+            or rename your file to match a supported filename pattern.
+            """
+        )
```

Comment:
I'm assuming the 16 indent is needed since we use `dals`, I think its better to not use `dals` then:

```suggestion
        msg = (
            f"No environment exporter plugin found for filename '{filename}'.\n"
            f"\n"
            f"Supported filenames: {dashlist(supported_filenames)}\n"
            f"\n"
            f"Available formats: {dashlist(available_formats)}\n"
            f"\n"
            f"Use `conda export --format=FORMAT` to specify the export format explicitly, "
            f"or rename your file to match a supported filename pattern."
        )
```

No strong opinion here just a suggestion 🤷🏼‍♂️